### PR TITLE
Update doc: SB3 Contrib RecurrentPPO

### PR DESCRIPTION
**Pull request by araffin**
## Description

Add Recurrent PPO to the list.

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)
- [ ] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: docs/guide/algos.rst [hunk 1]
```diff
@@ -15,6 +15,7 @@ DQN                  ❌           ✔️            ❌                ❌
 HER                  ✔️           ✔️            ❌                ❌               ❌
 PPO                  ✔️           ✔️            ✔️                 ✔️               ✔️
 QR-DQN [#f1]_        ❌          ️ ✔️            ❌                ❌               ✔️
+RecurrentPPO [#f1]_  ✔️           ✔️             ✔️                ✔️               ✔️
```
Comment (Miffyli) [inline docs/guide/algos.rst:18]: NItpick, but DQN and PPO variants could be next to their respective source algorithms (DQNs next to each other, PPOs next to each other).

File: docs/misc/changelog.rst [hunk 2]
```diff
@@ -18,6 +18,8 @@ New Features:
 
 SB3-Contrib
 ^^^^^^^^^^^
+- Added Recurrent PPO (PPO LSTM) see https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/pull/53
```
Comment (Miffyli) [inline docs/misc/changelog.rst:21]:
```suggestion
- Added Recurrent PPO (PPO LSTM). See https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/pull/53
```

File: docs/guide/algos.rst (hunk 1 above)
Comment (araffin) [reply docs/guide/algos.rst:18]: the order is alphabetical, so it's easier to search for something normally ;)

File: docs/modules/ppo.rst [hunk 3]
```diff
@@ -31,6 +31,14 @@ Notes
 Can I use?
 ----------
 
+.. note::
+
+  A recurrent version of PPO is available in our contrib repo: https://sb3-contrib.readthedocs.io/en/master/modules/ppo_recurrent.html
+
+  However we advise users to start with simple frame-stacking as a simpler, faster
+  and usually competitive alternative, more info in our report: https://wandb.ai/sb3/no-vel-envs/reports/PPO-vs-RecurrentPPO-aka-PPO-LSTM-on-environments-with-masked-velocity--VmlldzoxOTI4NjE4
```
Comment (Miffyli) [inline docs/modules/ppo.rst:39]: Could also point to Procgen paper's appendix with similar results (Fig 11): https://arxiv.org/abs/1912.01588

Also maybe a mention that you can do framestacking with `VecFrameStack`?

**Comment by Miffyli** [comment]
Few comments but LGTM otherwise :). Need to start using pyupgrade myself, too.

**Pull request merged**